articles/aws-api-setup.md

```diff
@@ -1,70 +1,104 @@
-  * **Attribute**: `SAML:iss`
-  * **Value**: `urn:${account.namespace}`
+  ![](/media/articles/aws-api-setup/aws-api-setup-7.png)
 
+4. Select or enter the following values:
 
-  ![](/media/articles/aws-api-setup/aws-api-setup-7.png)
+  * **SAML Provider**: the provider you just created
+  * **Attribute**: `SAML:iss`
+  * **Value**: `urn:${account.namespace}`
 
-4. This is just a confirmation. Click on **Continue**.
+  and click **Next Step**:
 
   ![](/media/articles/aws-api-setup/aws-api-setup-8.png)
 
-5. This is where you assign permissions to this role. In this example, we are creating a policy that give full access to the S3 resource `YOUR_BUCKET/<%= '${saml:sub}' %>`. This will be evaluated on runtime and replaced with the `user_id` of the logged in user.
+5. On the Verify Role Trust page, accept the Policy Document as provided. Click **Next Step**:
 
   ![](/media/articles/aws-api-setup/aws-api-setup-9.png)
 
-  > For more information about what AWS APIs are supported: <http://docs.aws.amazon.com/IAM/latest/UserGuide/Using_SpecificProducts.html>
+6.  Skip the **Attach Policy** step for now. Click **Next Step**.
 
-6. Confirm the creation of the new role.
+7. Review the information and click **Create Role**:
 
   ![](/media/articles/aws-api-setup/aws-api-setup-10.png)
 
-7. Copy the **Role ARN** value. You will use this value later when calling the `/delegation` endpoint in Auth0.
+## Create a Custom Policy
+
+In this example, you will create a policy that grants full access to the S3 resource: `YOUR_BUCKET/<%= '${saml:sub}' %>`. This will be evaluated at run-time and replaced with the `user_id` of the logged-in user.
+
+1. On the **Roles** page, select the role you just created.
+
+  On the **Permissions** tab under **Inline Policies**, click the link to create an inline policy:
 
   ![](/media/articles/aws-api-setup/aws-api-setup-11.png)
 
-8. Copy the **Principal ARN** value. You will use this value later when calling the `/delegation` endpoint in Auth0.
+3. Name your policy and enter the following code in the **Policy Document** field:
+
+  ```
+    {
+      "Version": "2012-10-17",
+      "Statement": [{
+          "Effect": "Allow",
+          "Action": [
+            "*"
+          ],
+          "Resource": [
+          "arn:aws:s3:::YOUR_BUCKET/$[saml:sub}",
+          "arn:aws:s3:::YOUR_BUCKET/$[saml:sub}/*"]
+      }]
+    }
+  ```
+
+4. Click **Validate Policy** to check your syntax, then click **Create Policy**.
 
   ![](/media/articles/aws-api-setup/aws-api-setup-12.png)
+
+## Copy the ARN values
+
+1. From the summary page of the role you just created, copy the **Role ARN** value. You will use this value later when calling the `/delegation` endpoint in Auth0.
+
+  ![](/media/articles/aws-api-setup/aws-api-setup-13.png)
+
+2. From the summary page of the identity provider you created previously, copy the **Provider ARN** value. You will use this value later when calling the `/delegation` endpoint in Auth0.
```

Comment:
That is the case. Ideally we would have called this Provider vs. Principal for naming, but since it is already in place, I was just going to fix this by documenting them as equivalent rather than changing the delegation code.